handler.go

```diff
@@ -86,3 +87,8 @@ func sendHeader(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 	}
 }
+
+func cacheHeaders(w http.ResponseWriter) {
+	w.Header().Add("Cache-Control", "max-age="+cacheTime)
+	w.Header().Add("Cache-Control", "public")
+}
```

Comment:
The Last-Modified will be a bit tricky as the data is generated and it is not clear to me what we would set it to. I was thinking about the ETag quite a bit and was not sure if it will be beneficial if we have a very long cache time. We could use the file sha1 as the etag for example.

Reply:
I think there are reasonable answers to those questions, but let's avoid them for now and focus on the `max-age` value.

I don't think the registry should use the same caching policy for all responses. 

`/package` seems well-suited for a "far future" expiration date. e.g. a year or more. The assets associated with a `package-x@major.minor.patch` should never change. AFAIK, the registry won't allow force push/republish, so any change should be a new version. That means the expiration should be able to be _well_ into the future without any issues.

`/search` & `/categories` are dynamic and should have the expirations set to whatever we feel is Fast Enough and/or doesn't overwhelm the registry.

Reply:
SGTM and agree on the different caching times. I expect we will have in the future a CDN in front so the total number of request on the service itself should become pretty low.